frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseCheckoutAction.java

```diff
@@ -0,0 +1,131 @@
+    super.onUpdate(anActionEvent);
+
+    val presentation = anActionEvent.getPresentation();
+    if (!presentation.isEnabledAndVisible()) {
+      return;
+    }
+
+    val toBeCheckedOutBranchName = getTargetBranchName(anActionEvent);
+
+    // It's very unlikely that selectedBranchName is empty at this point since it's assigned directly before invoking this
+    // action in GitMacheteGraphTable.GitMacheteGraphTableMouseAdapter.mouseClicked; still, it's better to be safe.
+    if (toBeCheckedOutBranchName == null || toBeCheckedOutBranchName.isEmpty()) {
+      presentation.setEnabled(false);
+      presentation.setDescription(getNonHtmlString("action.GitMachete.CheckoutSelectedAction.undefined.branch-name"));
+      return;
+    }
+
+    val currentBranchName = getCurrentBranchNameIfManaged(anActionEvent);
+
+    if (currentBranchName != null && currentBranchName.equals(toBeCheckedOutBranchName)) {
+      presentation.setEnabled(false);
+      presentation.setDescription(
+          getNonHtmlString("action.GitMachete.CheckoutSelectedAction.description.disabled.currently-checked-out")
+              .format(toBeCheckedOutBranchName));
+
+    } else {
+      presentation.setDescription(
+          getNonHtmlString("action.GitMachete.CheckoutSelectedAction.description.precise")
+              .format(toBeCheckedOutBranchName));
+    }
+  }
+
+  @Override
+  @UIEffect
+  public void actionPerformed(AnActionEvent anActionEvent) {
+    val toBeCheckedOutBranchName = getTargetBranchName(anActionEvent);
+    if (toBeCheckedOutBranchName == null || toBeCheckedOutBranchName.isEmpty()) {
+      return;
+    }
+
+    val project = getProject(anActionEvent);
+    val gitRepository = getSelectedGitRepository(anActionEvent);
+
+    if (gitRepository != null) {
+
+      log().debug(() -> "Queuing '${toBeCheckedOutBranchName}' branch checkout background task");
+      new Task.Backgroundable(project, getString("action.GitMachete.CheckoutSelectedAction.task-title")) {
+        @Override
+        @UIThreadUnsafe
+        public void run(ProgressIndicator indicator) {
+          doCheckout(project, indicator, toBeCheckedOutBranchName, gitRepository);
+        }
+      }.queue();
+    }
+  }
+
+  @UIThreadUnsafe
+  public static void doCheckout(Project project, @NonNull ProgressIndicator indicator, String branchToCheckoutName,
+      GitRepository gitRepository) {
+    val uiHandler = new GitBranchUiHandlerImpl(project, indicator);
+    new GitBranchWorker(project, Git.getInstance(), uiHandler)
+        .checkout(branchToCheckoutName, /* detach */ false, Collections.singletonList(gitRepository));
+  }
+
+  protected Stream<String> deriveLayoutBranches(AnActionEvent anActionEvent) {
+    val gitRepository = getSelectedGitRepository(anActionEvent);
+    if (gitRepository != null) {
+      val macheteFilePath = gitRepository.getMacheteFilePath();
+      try {
+        val macheteFileContent = Files.readString(macheteFilePath, StandardCharsets.UTF_8);
```

Comment:
I wouldn't operate directly on the file contents... pls use `BranchLayoutReader`, see e.g. `com.virtuslab.gitmachete.frontend.ui.impl.RediscoverSuggester`

Reply:
It produces the branchLayout. I tried that initially. The algorithm for jumping to the next branch using the branchLayout is recursive and longer, so I was unsure of keeping it.

1. see if the branch has children
2. if so go to the first child,
3. else see if the current branch is nonRoot,
4. if so, take the children of the parent,
5. find the next sibling,
6. if there is no sibling next, 
7. see if the parent is nonRoot
8. if so get the grandparent
9. get the children of the grandparent
10. find the next sibling of the parent
11. if there is no sibling for the parent, repeat from 7 for grandparent.

Reply:
The next alternative I would say is to add a method: `getNextBranch(branch)` to the branchLayout, and actually in the  branchLayout, either keep an ordered list of all branches in layout, or keep their order.

Reply:
It's just that in each update, we have to also update the list, which I think won't be that much of a deal.

Reply:
Hmmm lemme think... actually, the recursive algorithm will probably be cleaner than keeping extra piece of state in each node, just for the purpose of these actions.

I'd suggest the following:
* put this PR on hold for a moment,
* implement the algorithm of `getNext` and `getPrevious` in `BranchLayout` according to https://github.com/VirtusLab/git-machete-intellij-plugin/pull/1167#discussion_r982266997 on a separate PR, together with tests
* use the new methods to cleanly implement `{Previous,Next}BranchCheckoutAction`

WDYT?

Reply:
> implement the algorithm of getNext and getPrevious in BranchLayout 

using the recursive algorithm or the list of layout branches?

Reply:
> the recursive algorithm will probably be cleaner than keeping extra piece of state in each node

ok, got it. so I would go with recursive.